src/modules/demo_playback.rs

```diff
@@ -204,8 +204,18 @@ pub fn set_next_demo(marker: MainThreadMarker) {
                 cls_demos.demonum = 0;
 
                 let demo = demo.as_slice_of().unwrap();
-                cls_demos.demos[0][..demo.len()].copy_from_slice(demo);
-                cls_demos.demos[1][0] = 0;
+                let mut len = demo.len();
+                let mut count = 0;
+
+                // Assigning each row with sliced string
+                while len > 16 {
+                    cls_demos.demos[count].copy_from_slice(&demo[(count * 16)..((count + 1) * 16)]);
+                    len = len - 16;
+                    count = count + 1;
+                }
+                cls_demos.demos[count][..len]
+                    .copy_from_slice(&demo[(count * 16)..(count * 16 + len)]);
+                cls_demos.demos[count + 1][0] = 0;
```

Comment:
Not sure how necessary this line 218 is. I take it out of my build and it still works. This line could cause segfault.

Reply:
Thing is I don't think anyone would make a demo name 512 characters long so maybe this is fine until someone decides to just try it out.